Lib/typing.py

```diff
@@ -1189,6 +1290,10 @@ def _determine_new_args(self, args):
                                     f"ParamSpec, or Concatenate. Got {new_arg}")
             elif isinstance(old_arg, self._typevar_types):
                 new_arg = new_arg_by_param[old_arg]
+            elif (TypeVarTuple in self._typevar_types
+                  and _is_unpacked_typevartuple(old_arg)):
+                original_typevartuple = old_arg.__parameters__[0]
+                new_arg = new_arg_by_param[original_typevartuple]
```

Comment:
Seems this code is not covered by tests. All passed if remove it.

Reply:
Oops, good catch! Based on https://github.com/python/cpython/pull/31800 I guess you figured this out already, but for posterity: I think it's because this code path is currently masked off by the `NotImplementedError`.

I'm not sure https://github.com/python/cpython/pull/31800 is the full solution to this yet, but I'll try drafting an idea that Jelle proposed above (to return to a new `_GenericAlias` on subscription) now.